src/conf_mode/nat_cgnat.py

```diff
@@ -401,12 +414,38 @@ def generate(config):
 
 
 def apply(config):
-    if not config:
+    # Delete conntrack entries
+    # if the pool configuration has changed
+    if 'delete_conntrack_entries' in config and 'effective' in config:
+        # Prepare the list of internal pool prefixes
+        internal_pool_prefix_list: list[ipaddress.IPv4Network] = []
+
+        # Get effective rules configurations
+        for rule_config in config['effective'].get('rule', {}).values():
+            # Get effective internal pool configuration
+            internal_pool = rule_config['source']['pool']
+            # Find the internal IP ranges for the internal pool
+            internal_ip_ranges: list[str] = config['effective']['pool']['internal'][
+                internal_pool
+            ]['range']
+            # Get the IP prefixes for the internal IP range
+            for internal_range in internal_ip_ranges:
+                ip_prefix: list[ipaddress.IPv4Network] = IPOperations(
+                    internal_range
+                ).get_prefix_by_ip_range()
+                # Add the IP prefixes to the list of all internal pool prefixes
+                internal_pool_prefix_list += ip_prefix
+
+        # Delete required sources for conntrack
+        _delete_conntrack_entries(internal_pool_prefix_list)
```

Comment:
The previous location was proper - we must clean up _after_ changing the nftables configuration.

Reply:
> The previous location was proper - we must clean up _after_ changing the nftables configuration.

If we do not have config, i.e. `remove` in config, It will return `None`, and the code with cleanup conntrack entries will never be executed.

This way we cleanup the conntrack entries and do nothing `return None`